### PR TITLE
Finalizing SP metadata upload to IS

### PR DESCRIPTION
**Pull request by PasinduTennage**

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 1]
```diff
@@ -82,6 +86,28 @@
             <groupId>org.wso2.carbon.commons</groupId>
             <artifactId>org.wso2.carbon.tenant.common</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity</groupId>
+            <artifactId>org.wso2.carbon.identity.saml.metadata</artifactId>
+            <version>5.1.2-SNAPSHOT</version>
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: define a property for the version at the root level and refer to it here.

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 2]
```diff
@@ -82,6 +86,28 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.wso2.carbon.identity</groupId>
+            <artifactId>org.wso2.carbon.identity.sso.saml</artifactId>
+            <version>5.1.2-SNAPSHOT</version>
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: define the version as a property

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/persistence/IdentityPersistenceManager.java [hunk 3]
```diff
@@ -240,6 +237,17 @@ public boolean addServiceProvider(Registry registry, SAMLSSOServiceProviderDO se
...
         return serviceProviderDAO.addServiceProvider(serviceProviderDO);
     }
+    /**
+     * Upload Service Provider
+     *
+     * @param registry,samlssoServiceProviderDO
+     * @return
+     * @throws IdentityException
+     */
+    public SAMLSSOServiceProviderDO uploadServiceProvider(Registry registry, SAMLSSOServiceProviderDO samlssoServiceProviderDO) throws IdentityException {
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/persistence/IdentityPersistenceManager.java]: stick to 120 characters limit.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/IdentityCoreServiceComponent.java [hunk 4]
```diff
@@ -78,6 +102,7 @@ protected void activate(ComponentContext ctxt) {
             log.debug("Identity Core bundle is activated");
         }
         try {
+//            DefaultBootstrap.bootstrap();
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/IdentityCoreServiceComponent.java]: Remove commented code and the import.

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 5]
```diff
@@ -82,6 +86,28 @@
...
+        </dependency>
+
+
+        <dependency>
+            <groupId>org.wso2.carbon.identity.inbound.auth.saml2</groupId>
+            <artifactId>org.wso2.carbon.identity.sso.saml.common</artifactId>
+            <version>5.1.3-SNAPSHOT</version>
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: define version as property.

**Comment by PasinduTennage** [comment]
Closing this PR because these fixes are available in 
https://github.com/wso2/carbon-identity-framework/pull/602

**Pull request closed without merging**